### PR TITLE
perf: Improve recurrence service filtering

### PR DESCRIPTION
**Pull request by taratatach**
We can skip transactions with a recurrence or marked as not recurrent
via a partial filter instead of requesting all the transactions first
and then loop over them to remove those that we don't want to modify.

```
### 🔧 Tech

* Improved performance of transactions filtering in the recurrence service
```

### CODE REVIEW

**Review by nono — CHANGES_REQUESTED** [review]
Are you really sure about this change? I don't know cozy-client syntax, but it looks like the filtering on the date is not longer done at indexing time, but at query time, and if it is the case, it would be a bad regression for performances.

File: src/ducks/recurrence/service.js
Comment (paultranvan) [inline src/ducks/recurrence/service.js:35]: You may want to specify a `.limit(1000)`, otherwise, the pagination will be 100 by default, resulting in ~10x more queries

Comment (paultranvan) [inline src/ducks/recurrence/service.js:35]: For this particular case, I'm not sure the partialFilter is actually worth it: this will create an index for probably few docs (the ones marked as not recurrent), and will use the `_find` endpoint instead of the `_all_docs` endpoint. And the later is actually more efficient

**Comment by nono** [comment]
It looks like I didn't understand the index/query made by cozy-client.

File: src/ducks/recurrence/service.js
Comment (taratatach) [reply src/ducks/recurrence/service.js:35]: As I don't have any information on the number of marked documents I guess it's a good bet to remove the partial index but we might have more marked documents than we think when there aren't any recurrence. Especially if the user deleted all their recurrences (all associated transactions are then marked as not recurrent).

Comment (taratatach) [reply src/ducks/recurrence/service.js:35]: Ah, one more thing : removing the partial filter here means we'll need to loop over the fetched transactions to skip the few ones marked as not recurrent :/

**Review by Crash-- — APPROVED** [review]
Can you squash the two commits before rebase & merge ? 🙏

**Comment by taratatach** [comment]
Yes, I never leave fixups around :).

File: src/ducks/recurrence/service.js
Comment (paultranvan) [reply src/ducks/recurrence/service.js]: Well, if we can't estimate the potential number of marked documents, and therefore the actual impact, I would say it's your call :)
I tend to think the average case would be in favor of removing the partial filter, but I don't have any data to support this, and the need to loop over all the transactions afterwards to filter out the recurrences is cumbersome. So either way, it's fine by me

**Review by paultranvan — APPROVED** [review]

**Pull request merged**
